src/java/org/apache/cassandra/db/virtual/SystemViewsKeyspace.java

```diff
@@ -31,27 +31,35 @@
 {
+    private static final boolean ONLY_LOCAL_AND_PEERS = Boolean.getBoolean("cassandra.system_view.only_local_and_peers_table");
+
     public static SystemViewsKeyspace instance = new SystemViewsKeyspace();
 
     private SystemViewsKeyspace()
     {
-        super(VIRTUAL_VIEWS, new ImmutableList.Builder<VirtualTable>()
-                    .add(new CachesTable(VIRTUAL_VIEWS))
-                    .add(new ClientsTable(VIRTUAL_VIEWS))
-                    .add(new SettingsTable(VIRTUAL_VIEWS))
-                    .add(new SystemPropertiesTable(VIRTUAL_VIEWS))
-                    .add(new SSTableTasksTable(VIRTUAL_VIEWS))
-                    .add(new ThreadPoolsTable(VIRTUAL_VIEWS))
-                    .add(new InternodeOutboundTable(VIRTUAL_VIEWS))
-                    .add(new InternodeInboundTable(VIRTUAL_VIEWS))
-                    .add(new SSTablesSystemView(VIRTUAL_VIEWS))
-                    .add(new SegmentsSystemView(VIRTUAL_VIEWS))
-                    .add(new IndexesSystemView(VIRTUAL_VIEWS))
-                    .add(new AnalyzerView(VIRTUAL_VIEWS))
-                    .addAll(TableMetricTables.getAll(VIRTUAL_VIEWS))
-                    .add(new LocalNodeSystemView())
-                    .add(new PeersSystemView())
-                    .add(new LegacyPeersSystemView())
-                    .build());
+        super(VIRTUAL_VIEWS, ONLY_LOCAL_AND_PEERS ?
+                             new ImmutableList.Builder<VirtualTable>()
+                             .add(new LocalNodeSystemView())
+                             .add(new PeersSystemView())
+                             .add(new LegacyPeersSystemView())
+                             .build() :
+                             new ImmutableList.Builder<VirtualTable>()
+                             .add(new CachesTable(VIRTUAL_VIEWS))
+                             .add(new ClientsTable(VIRTUAL_VIEWS))
+                             .add(new SettingsTable(VIRTUAL_VIEWS))
+                             .add(new SystemPropertiesTable(VIRTUAL_VIEWS))
+                             .add(new SSTableTasksTable(VIRTUAL_VIEWS))
+                             .add(new ThreadPoolsTable(VIRTUAL_VIEWS))
+                             .add(new InternodeOutboundTable(VIRTUAL_VIEWS))
+                             .add(new InternodeInboundTable(VIRTUAL_VIEWS))
+                             .add(new SSTablesSystemView(VIRTUAL_VIEWS))
+                             .add(new SegmentsSystemView(VIRTUAL_VIEWS))
+                             .add(new IndexesSystemView(VIRTUAL_VIEWS))
+                             .add(new AnalyzerView(VIRTUAL_VIEWS))
+                             .addAll(TableMetricTables.getAll(VIRTUAL_VIEWS))
+                             .add(new LocalNodeSystemView())
+                             .add(new PeersSystemView())
+                             .add(new LegacyPeersSystemView())
+                             .build());
```

Comment:
Tow nits here. 
1. [Apache Cassandra code style](https://cassandra.apache.org/_/development/code_style.html) specifies to carry operators on new line.
2. It would be nice to improve readability by utilising C* indentation.
I see the following way for improvement, unfortunately at the cost of longer lines:
```suggestion
        super(VIRTUAL_VIEWS, ONLY_LOCAL_AND_PEERS
                             ? new ImmutableList.Builder<VirtualTable>().add(new LocalNodeSystemView())
                                                                        .add(new PeersSystemView())
                                                                        .add(new LegacyPeersSystemView())
                                                                        .build()
                             : new ImmutableList.Builder<VirtualTable>().add(new CachesTable(VIRTUAL_VIEWS))
                                                                        .add(new ClientsTable(VIRTUAL_VIEWS))
                                                                        .add(new SettingsTable(VIRTUAL_VIEWS))
                                                                        .add(new SystemPropertiesTable(VIRTUAL_VIEWS))
                                                                        .add(new SSTableTasksTable(VIRTUAL_VIEWS))
                                                                        .add(new ThreadPoolsTable(VIRTUAL_VIEWS))
                                                                        .add(new InternodeOutboundTable(VIRTUAL_VIEWS))
                                                                        .add(new InternodeInboundTable(VIRTUAL_VIEWS))
                                                                        .add(new SSTablesSystemView(VIRTUAL_VIEWS))
                                                                        .add(new SegmentsSystemView(VIRTUAL_VIEWS))
                                                                        .add(new IndexesSystemView(VIRTUAL_VIEWS))
                                                                        .add(new AnalyzerView(VIRTUAL_VIEWS))
                                                                        .addAll(TableMetricTables.getAll(VIRTUAL_VIEWS))
                                                                        .add(new LocalNodeSystemView())
                                                                        .add(new PeersSystemView())
                                                                        .add(new LegacyPeersSystemView())
                                                                        .build());
```

Reply:
Or may be avoid repeating to construct local tables and have something like:
```java
    private static Collection<VirtualTable> buildTables()
    {
        ImmutableList.Builder<VirtualTable> tables = new ImmutableList.Builder<VirtualTable>();
        if (!ONLY_LOCAL_AND_PEERS)
            tables.add(new CachesTable(VIRTUAL_VIEWS))
                  .add(new ClientsTable(VIRTUAL_VIEWS))
                  .add(new SettingsTable(VIRTUAL_VIEWS))
                  .add(new SystemPropertiesTable(VIRTUAL_VIEWS))
                  .add(new SSTableTasksTable(VIRTUAL_VIEWS))
                  .add(new ThreadPoolsTable(VIRTUAL_VIEWS))
                  .add(new InternodeOutboundTable(VIRTUAL_VIEWS))
                  .add(new InternodeInboundTable(VIRTUAL_VIEWS))
                  .add(new SSTablesSystemView(VIRTUAL_VIEWS))
                  .add(new SegmentsSystemView(VIRTUAL_VIEWS))
                  .add(new IndexesSystemView(VIRTUAL_VIEWS))
                  .add(new AnalyzerView(VIRTUAL_VIEWS))
                  .addAll(TableMetricTables.getAll(VIRTUAL_VIEWS));
        tables.add(new LocalNodeSystemView())
              .add(new PeersSystemView())
              .add(new LegacyPeersSystemView());
        return tables.build();
    }
    private SystemViewsKeyspace()
    {
        super(VIRTUAL_VIEWS, buildTables());
    }
```

Reply:
I prefer this one. I never liked the original from DSE but didn't give it much thought. I'll get this in.